README.md

```diff
@@ -1 +1,922 @@
+
+## Making the source available to the build environment
+
+### Source tarballs via a mirror
+
+#### Mirrors:
+
+The omnios-extra build system has a mirror at https://mirrors.omniosce.org/ that maintains current source tarballs and checksums. It uses these to build it's packages. In the early stages of a build, this source tarball will not be available on the OmniOS mirror, and will be need to be made available for the build process.
+
+This can be done by using the `set_mirror` directive in `build.sh`. For example the source tarball for "Apache httpd 2.4.43" is available at the mirror: <https://downloads.apache.org/httpd/>. Therefore the `set_mirror` directive should be as follows:
+
+```none
+set_mirror "https://downloads.apache.org/httpd/"
+```
+Further in the `build.sh` file, the `download_source` directive should be as follows:
+
+```none
+download_source $PROG $PROG $VER
+```
+
+#### Checksums:
+
+The above will be sufficient if the accompanying sha256 checksum file is available in the same directory as the tarball. 
+
+If the checksum is not available, this can be included in the `build.sh` with the `set_checksum` directive. For example, had the "Apache httpd 2.4.43" checksum file not been available, the following `set_checksum` should be set:
+
+```none
+set_checksum "a497652ab3fc81318cdc2a203090a999150d86461acff97c1065dc910fe10f43"
+```
+
+### Source tarballs via GitHub
+
+#### GitHub releases:
+
+If the source tarball is on github, the following directive should be used:
+
+```none
+set_mirror "$GITHUB/apache/$PROG/archive"
+```
+
+Further in the `build.sh` file, the `download_source` directive should be as follows:
+
+```none
+download_source $VER
+```
+
+> **NOTE:** Downloading from GitHub always seems to be a moving target. See the [Exceptional cases regarding source tarballs](#exceptional-cases-regarding-source-tarballs) section to help troubleshoot `download_source` problems.
+
+#### Checksums:
+
+GitHub does not supply checksums for downloaded archives, so this can be workd around with the following directive:
+
+```none
+set_checksum "none"
+```
+
+### Exceptional cases regarding source tarballs
+
+
+#### Source tarball
+
+Sometimes the source tarball is not in a standard format and the argument to the directive, `download_source`, will need to be edited accordingly.
+
+#### Build Directory
+
+Sometimes the tarball will extract to an un-expected directory. For example, sometimes a tarball named `$PROG-$VER.tar.gz` may extract to a directory named `$PROG-$PROG-$VER`.
+
+In this case, use the `BUILDDIR` directive, as follows:
+
+```none
+BUILDIR="$PROG-$PROG-$VER" 
+```
+
+#### Source Directory
+
+Sometimes the source directory, where the `configure` script lies will not be where it is expected. For example, the `configure` script may not be in the root of the `$PROG-VER/` directory, but instead in `$PROG-VER/$PROG/`.
+
+In this case, use the `“EXTRACTED_SRC` directive, as follows:
+
+```none
+EXTRACTED_SRC+="/$PROG"` 
```

Comment:
You should never need to do this. `set_builddir`, see above, takes care of EXTRACTED_SRC too